addons/wakatime/wakatime.gd

```diff
@@ -3,61 +3,295 @@ extends EditorPlugin
 
 const HeartBeat = preload('res://addons/wakatime/heartbeat.gd')
 const Settings = preload('res://addons/wakatime/settings.gd')
-const Utils = preload('res://addons/wakatime/utils.gd')
+
+const PLUGIN_PATH = 'res://addons/wakatime'
+const WAKATIME_ZIP_FILEPATH = '%s/wakatime.zip' % PLUGIN_PATH
+
+const WAKATIME_URL_FMT = 'https://github.com/wakatime/wakatime-cli/releases/download/v1.54.0/{wakatime_build}.zip'
+const DECOMPRESSOR_URL_FMT = 'https://github.com/ouch-org/ouch/releases/download/0.3.1/{ouch_build}'
 
 var last_heartbeat = HeartBeat.new()
 
-var wakatime_cli = ProjectSettings.globalize_path(Utils.WAKATIME_CLI_PATH)
+var wakatime_dir = ''
+var wakatime_cli = ''
+var decompressor_cli = ''
 
 var api_key_modal = preload('res://addons/wakatime/api_key_modal.tscn')
 var bottom_panel_scn = preload('res://addons/wakatime/bottom_panel.tscn')
 var bottom_panel = null
 var settings = null
 
+var is_windows = OS.has_feature('windows') or OS.has_feature('uwp')
+var is_linux = OS.has_feature('linux')
+var is_macos = OS.has_feature('macos')
+var is_amd64 = OS.has_feature('x86_64')
+var is_arm64 = OS.has_feature('arm64')
 
```

Comment:
This may be working in the primary master branch, but since it didn't work for me, an investigation shows that they are lower case and case sensitive (at least as of V4B6).